core/eval.go

```diff
@@ -230,6 +232,64 @@ func evalMSET(args []string) []byte {
 	return RespOK
 }
 
+// evalSCAN processes the SCAN command with the given arguments.
+// It scans the database for keys based on the provided cursor, pattern, count, and key type.
+// Arguments:
+// - args: The command arguments, where args[0] is the cursor and optional args provide MATCH, COUNT, and TYPE options.
+// Returns:
+// - Encoded RESP value with:
+//   - The new cursor position as a string
+//   - A slice of keys that match the criteria
+//
+// If there are any errors in the arguments, an error message is returned.
+func evalSCAN(args []string) []byte {
+	// Check if there is at least one argument (cursor)
+	if len(args) < 1 {
+		return Encode(errors.New("ERR wrong number of arguments for 'scan' command"), false)
+	}
+
+	// Parse the cursor from the first argument
+	cursor, err := strconv.Atoi(args[0])
+	if err != nil {
+		return Encode(errors.New("ERR invalid cursor"), false)
+	}
+
```

Comment:
we must ensure that cursor value is >= 0 .

Reply:
Yes, will add that check